google/cloud/aiplatform/models.py

```diff
@@ -215,10 +228,534 @@ def upload(
+            parent (str):
+                Required. Also known as common location path, that usually contains
+                the project and location that the user provided to the upstream
+                method.
+                Example: "projects/my-prj/locations/us-central1"
+            display_name (str):
+                Required. The user-defined name of the Endpoint.
+                The name can be up to 128 characters long and can be consist
+                of any UTF-8 characters.
+            description (str):
+                Optional. The description of the Endpoint.
+            labels (Dict):
+                Optional. The labels with user-defined metadata to
+                organize your Endpoints.
+                Label keys and values can be no longer than 64
+                characters (Unicode codepoints), can only
+                contain lowercase letters, numeric characters,
+                underscores and dashes. International characters
+                are allowed.
+                See https://goo.gl/xmQnxf for more information
+                and examples of labels.
+            metadata (Sequence[Tuple[str, str]]):
+                Optional. Strings which should be sent along with the request as
+                metadata.
+        Returns:
+            operation (lro.LRO):
+                Long-running operation of endpoint creation.
+        """
+        gapic_endpoint = gca_endpoint.Endpoint(
+            display_name=display_name, description=description, labels=labels,
+        )
+
+        operation_future = api_client.create_endpoint(
+            parent=parent, endpoint=gapic_endpoint, metadata=metadata
+        )
+
+        return lro.LRO(operation_future)
+
+    def _allocate_traffic(
+        traffic_split: Dict[str, int], traffic_percentage: int,
+    ) -> Dict[str, int]:
+        """
+        Allocates desired traffic to new deployed model and scales traffic of
+        older deployed models.
+
+        Args:
+            traffic_split (Dict[str, int]):
+                Required. Current traffic split of deployed models in endpoint.
+            traffic_percentage (int):
+                Required. Desired traffic to new deployed model.
+        Returns:
+            new_traffic_split (Dict[str, int]):
+                Traffic split to use.
+        """
+        new_traffic_split = {}
+        old_models_traffic = 100 - traffic_percentage
+        if old_models_traffic:
+            unallocated_traffic = old_models_traffic
+            for deployed_model in traffic_split:
+                current_traffic = traffic_split[deployed_model]
+                new_traffic = int(current_traffic / 100 * old_models_traffic)
+                new_traffic_split[deployed_model] = new_traffic
+                unallocated_traffic -= new_traffic
+            # will likely under-allocate. make total 100.
+            for deployed_model in new_traffic_split:
```

Comment:
Will `unallocated_traffic` <= len(deployed_models) always be True?

Reply:
it should. int/floor/truncating a number in division should result in <1 difference per number.